code/modules/mob/living/carbon/human/say.dm

```diff
@@ -5,29 +5,30 @@
 		verb_ask = pick(T.ask_mod)
 		verb_yell = pick(T.yell_mod)
 		verb_exclaim = pick(T.exclaim_mod)
+	if(wear_mask && istype(wear_mask, /obj/item/clothing/mask))
+		var/obj/item/clothing/mask/worn_mask = wear_mask
+		if(!isnull(worn_mask.chosen_tongue))
+			verb_say = pick(initial(worn_mask.chosen_tongue.say_mod))
+			verb_ask = pick(initial(worn_mask.chosen_tongue.ask_mod))
+			verb_yell = pick(initial(worn_mask.chosen_tongue.yell_mod))
+			verb_exclaim = pick(initial(worn_mask.chosen_tongue.exclaim_mod))
 	if(slurring || !T)
 		return "slurs"
 	else
 		. = ..()
 
 /mob/living/carbon/human/GetVoice()
-	if(istype(wear_mask, /obj/item/clothing/mask/chameleon))
-		var/obj/item/clothing/mask/chameleon/V = wear_mask
-		if(V.vchange && wear_id)
-			var/obj/item/card/id/idcard = wear_id.GetID()
-			if(istype(idcard) && idcard.electric)
-				return idcard.registered_name
-			else
-				return real_name
-		else
-			return real_name
+	var/current_name = real_name
+	if(GetSpecialVoice())
+		current_name = GetSpecialVoice()
 	if(mind)
 		var/datum/antagonist/changeling/changeling = mind.has_antag_datum(/datum/antagonist/changeling)
 		if(changeling && changeling.mimicing )
-			return changeling.mimicing
-	if(GetSpecialVoice())
-		return GetSpecialVoice()
-	return real_name
+			current_name = changeling.mimicing
+	if(wear_mask && istype(wear_mask, /obj/item/clothing/mask))
+		var/obj/item/clothing/mask/gas/old/modulator/modulator = wear_mask
```

Comment:
you are casting into a type which isnt checked, only /obj/item/clothing/mask is checked so the most you can type this variable as is `/obj/item/clothing/mask`.
Either change the type of the variable to `/obj/item/clothing/mask` (if the get_name proc is on mask) or make the istype stricter.